src/auth/awss3/core/qgsauthawss3method.cpp

```diff
@@ -134,6 +137,85 @@ bool QgsAuthAwsS3Method::updateNetworkRequest( QNetworkRequest &request, const Q
   return true;
 }
 
+bool QgsAuthAwsS3Method::updateDataSourceUriItems( QStringList &connectionItems, const QString &authcfg,
+    const QString &dataprovider )
+{
+  Q_UNUSED( dataprovider )
+  const QMutexLocker locker( &mMutex );
+  const QgsAuthMethodConfig config = getMethodConfig( authcfg );
+  if ( !config.isValid() )
+  {
+    QgsDebugError( QStringLiteral( "Update URI items FAILED for authcfg: %1: basic config invalid" ).arg( authcfg ) );
+    return false;
+  }
+
+  if ( connectionItems.isEmpty() )
+  {
+    QgsDebugError( QStringLiteral( "Connection Items is empty!" ) );
+    return false;
+  }
+
+  const QByteArray username = config.config( QStringLiteral( "username" ) ).toLocal8Bit();
+  const QByteArray password = config.config( QStringLiteral( "password" ) ).toLocal8Bit();
+  const QByteArray region = config.config( QStringLiteral( "region" ) ).toLocal8Bit();
+
+  const QByteArray headerList = "host";
+  const QByteArray encryptionMethod = "AWS4-HMAC-SHA256";
+  const QDateTime currentDateTime = QDateTime::currentDateTime().toUTC();
+  const QByteArray date = currentDateTime.toString( "yyyyMMdd" ).toLocal8Bit();
+  const QByteArray dateTime = currentDateTime.toString( "yyyyMMddThhmmssZ" ).toLocal8Bit();
+  const QByteArray expires = "300";
+  const QByteArray method = "GET";
+
+  const QString fullUri( connectionItems.first() );
+  QString uri( fullUri );
+
+  QByteArray canonicalPath = QUrl::toPercentEncoding( QUrl(uri).path(), "/" );
+  if ( canonicalPath.isEmpty() )
+  {
+    canonicalPath = "/";
+  }
+
+  const QByteArray canonicalRequest = method + '\n' +
```

Comment:
In short, yes. The reason is that it needs to reflect the actual http request that gets sent, and in this case because parameters are being added as part of the query string and not the headers.

the s3 service creates its own canonicalrequest based upon the http request that it uses for signature verification, and so they need to match.